aten/src/ATen/native/EmbeddingBag.cpp

```diff
@@ -84,13 +168,25 @@ static void index_select_scale_add(const Tensor &select_indices,
   auto* scale_data = scale.data<T>();
   auto scale_stride = scale.stride(0);
 
-  // XXX: We could make this faster via vectorization
-  for (int64_t i = 0; i < numel; i++) {
-    auto* src_base = src_data + src_stride0 * select_indices_data[i];
-    auto* output_base = output_data + output_stride0 * add_indices_data[i];
-    auto scale = scale_data[i * scale_stride];
-    for (int64_t j = 0; j < ddim; j++) {
-      output_base[j * output_stride1] += src_base[j * src_stride1] * scale;
+  // We specialize the case where both the src and output are contiguous in the
+  // last dimension to give the compiler a better chance to vectorize.
+  if (src_stride1 == 1 && output_stride1 == 1) {
+    for (int64_t i = 0; i < numel; i++) {
```

Comment:
EmbeddingLookup supports weighted version too - see `weights` argument, so let's call it here